source/common/network/listener_impl.cc

```diff
@@ -98,10 +115,11 @@ void ListenerImpl::setupServerSocket(Event::DispatcherImpl& dispatcher, Socket&
 }
 
 ListenerImpl::ListenerImpl(Event::DispatcherImpl& dispatcher, SocketSharedPtr socket,
-                           ListenerCallbacks& cb, bool bind_to_port)
+                           ListenerCallbacks& cb, bool bind_to_port, const std::string& name)
     : BaseListenerImpl(dispatcher, std::move(socket)), cb_(cb), listener_(nullptr) {
   if (bind_to_port) {
     setupServerSocket(dispatcher, *socket_);
+    setupPipeListener(dispatcher, name);
```

Comment:
@mattklein123 
I registered each listener as internal connectable. This will lead to failure in the follow up.
I incline to create a new InternalListener type so that I can bootstrap the new type of listener,
adding very limited feature at the early phase. 

WDYT?